x/tokenregistry/keeper/keeper.go

```diff
@@ -120,10 +97,52 @@ func (k keeper) RemoveToken(ctx sdk.Context, denom string) {
 	store.Delete(key)
 }
 
+// GetRegistryEntry to get a token from the Token Registry without looping
+// Returns Error if entry is not found, Panics if entry is found but unable to be Unmarshalled
+func (k keeper) GetRegistryEntry(ctx sdk.Context, denom string) (*types.RegistryEntry, error) {
```

Comment:
Opinion:
I think we should have an additional return boolean indicating if entry is found, I see code relying on error != nil to do that, which we might not want, if down the road we want to also throw other kinds of error
Not required for this PR

Reply:
Isn't not found a type of error, we just would throw a different error type and you would have a handler for each error type. Code that works on not found I tends to be handling a get failure regardless of why it happened.

Reply:
As long as the caller has handler for each error type. My main concern is they'll just check err != nil and infer it's not found, when it could've been because their request is malformed. Now if their logic is to create a new entry when not found, that could cause an issue.